indent/ps1.vim

```diff
@@ -1,20 +1,96 @@
 " Vim indent file
-" Language:           Windows PowerShell
-" Maintainer:         Peter Provost <peter@provost.org>
-" Version:            2.10
-" Project Repository: https://github.com/PProvost/vim-ps1
-" Vim Script Page:    http://www.vim.org/scripts/script.php?script_id=1327"
+" Language:		PowerShell
+" Maintainer:	Lior Elia - http://blogs.msdn.com/lior
+" Version:		2.0
+" Last Change:	2009 Oct 18
+" Changelist:
+" 	Ver		Date		By			Fixed indentation of...
+" 	-----	----------	--------	-----------------------------------------
+"	1.0		2009-10-04	LiorE		Comment rows
+" 	2.0		2009-10-18	LiorE		Lines after (.*), comments after '{' or
+"					 				'}' and many many other things. I've made
+"					 				this a major version change because there
+"					 				are many changes in the behavior.
+
+" Dedicated to my loving and supporting family: Anat, Amit and Ofir.
 
 " Only load this indent file when no other was loaded.
 if exists("b:did_indent")
 	finish
 endif
 let b:did_indent = 1
 
-" smartindent is good enough for powershell
-setlocal smartindent
-" disable the indent removal for # marks
-inoremap # X#
+setlocal cindent cinoptions& cinoptions+=+0
+
+" Set the function to do the work.
+setlocal indentexpr=GetPsIndent()
+
+let b:undo_indent = "set cin< cino< indentkeys< indentexpr<"
+
+" Only define the function once.
+if exists("*GetPsIndent")
+	finish
+endif
+
+function! Log(text)
```

Comment:
No reason to include dead code. If someone would like to debug, VIM supports built-in mechanisms or can add tracing themselves. It should not be included in shipping code.